### DISCUSSION TITLE
build error with 1.19.0 (async branch)

### DISCUSSION THREAD

**Discussion by bhopmann** [dani-garcia/vaultwarden#1370] [Q&A]
### Subject of the issue

Latest `1.19.0` version on `async` branch fails to build.

### Your environment

* Bitwarden_rs version: `1.19.0` (using the `async` branch):

* Install method: manual build
* Clients used: 
* Reverse proxy and version: 
* Version of mysql/postgresql: 
* Other relevant information: compile for sqlite 

### Steps to reproduce

```
git clone https://github.com/dani-garcia/bitwarden_rs.git
git checkout origin/async
cargo build -j 1 --release --features sqlite
```

### Expected behaviour

successful build

### Actual behaviour

build error when compiling `bitwarden_rs`

### Relevant logs

```
error: could not compile `bitwarden_rs`

Caused by:
  process didn't exit successfully:
[...]
(signal: 9, SIGKILL: kill)
```

**Comment by BlackDex** [comment]
On what kind of hardware are you trying to build this?
Most of the time these errors are low memory related.

Also, the async branch could be build with the latest stable if im correct. You just need to override the rust-toolchain file by using `cargo +stable build ...`

**Reply by bhopmann** [reply]
I'm using shared hosting on https://uberspace.de

**Reply by BlackDex** [reply]
And what is the amount of memory you have access to?

**Reply by bhopmann** [reply]
I had to investigate a little bit: Uberspace seems to has a memory limit of around 1.5 GiB (see https://manual.uberspace.de/basics-resources/#ram)

Edit: Version `1.18.0` compiled without any (memory) problems

**Reply by dani-garcia** [reply]
Building the latest async branch with rust 1.49.0 for me works, but it seems to peak at 1.7GB of RAM on my x86-64 PC, I assume we were pretty close to the limit before, and an updated dependency might just have gone over the limit, so it doesn't work now.

As a workaround, compiling without the --release flag uses a bit less resources in my PC, around 1.2 GB of RAM at peak.

**Reply by bhopmann** [reply]
Thank you for your help @BlackDex and @dani-garcia - I've stopped other processes and now I'm able to build this version successfully.

By the way: When trying to open my previously added organisation settings, bitwarden_rs crashes with the following message:

```
thread 'tokio-runtime-worker' has overflowed its stack
fatal runtime error: stack overflow
```

May this be also related to a memory failure? Or is it caused by something else?

**Reply by dani-garcia** [reply]
Oops, that was my fault! That's what I get for copy pasting when trying to rebase all the commits lol, should be fixed now!

**Reply by bhopmann** [reply]
Thanks :-) It's working now!
